py/picca/delta_extraction/masks/bal_mask.py

```diff
@@ -157,50 +163,40 @@ def add_bal_rest_frame(self, los_id, los_id_name):
         los_id_name: str
         Name of the line-of-sight id
         """
-        if self.bal_index_type == 'bi':
-            velocity_list = ['VMIN_CIV_2000', 'VMAX_CIV_2000']
-        else:  # AI, the default
-            velocity_list = ['VMIN_CIV_450', 'VMAX_CIV_450']
-
-        mask_rest_frame_bal = Table(names=[
-            'log_lambda_min', 'log_lambda_max', 'lambda_min', 'lambda_max'
-        ],
-                                    dtype=['f4', 'f4', 'f4', 'f4'])
-        min_velocities = []  # list of minimum velocities
-        max_velocities = []  # list of maximum velocities
-
         # Match thing_id of object to BAL catalog index
-        match_index = np.where(self.cat[los_id_name] == los_id)[0][0]
+        # Will there be duplicates or only one index?
+        match_index = np.where(self.cat[los_id_name] == los_id)[0]
 
         # Store the min/max velocity pairs from the BAL catalog
-        for col in velocity_list:
-            if col.find('VMIN') == 0:
-                velocity_list = self.cat[col]
-                for vel in velocity_list[match_index]:
-                    if vel > 0:
-                        min_velocities.append(vel)
-            else:
-                velocity_list = self.cat[col]
-                for vel in velocity_list[match_index]:
-                    if vel > 0:
-                        max_velocities.append(vel)
+        colmin = self.cat[self.velocity_list[0]]
+        colmax = self.cat[self.velocity_list[1]]
+        # np.array of minimum velocities
+        min_velocities = np.array(colmin[match_index], dtype=float)
+        # np.array of maximum velocities
+        max_velocities = np.array(colmax[match_index], dtype=float)
+        # Remove non-positive velocity rows
+        w = (min_velocities>0) & (max_velocities>0)
+        min_velocities = min_velocities[w]
+        max_velocities = max_velocities[w]
+
+        nvelocities = w.sum()
```

Comment:
I'd do `min_velocities.size` so that it is more clear. Maybe also rename `nvelocities` to `num_velocities` and `nlines` to `num_lines`